stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -557,6 +557,19 @@ julia> exp(A)
 exp(A::StridedMatrix{<:BlasFloat}) = exp!(copy(A))
 exp(A::StridedMatrix{<:Union{Integer,Complex{<:Integer}}}) = exp!(float.(A))
 
+"""
+    cis(A::AbstractMatrix)
+
+Compute ``\\exp(i A)`` for a generic matrix ``A``.
+
+# Examples
+```jldoctest
+julia> cis([π 0; 0 π]) ≈ -I
+true
+```
+"""
+Base.cis(A::AbstractMatrix) = exp(im * A)  # fallback
```

Comment:
Should we use exp! for appropriate matrix types?

Reply:
Do you mean simply `Base.cis(A::AbstractMatrix) = exp!(im .* A)`?

Reply:
Not quite.  `exp!` only works for certain matrix types, so we might need something like:
```jl
Base.cis(A::AbstractMatrix) = exp_maybe_inplace(float.(im .* A))
exp_maybe_inplace(A::StridedMatrix{T}) where T<:Union{Float32, Float64, ComplexF32, ComplexF64} = exp!(A)
exp_maybe_inplace(A) = exp(A)
```

Reply:
Perfect, done!